{"ast":null,"code":"import { catchError, throwError } from 'rxjs';\nimport * as i0 from \"@angular/core\";\nimport * as i1 from \"@angular/common/http\";\nexport class UserServices {\n  constructor(http) {\n    this.http = http;\n  }\n  getRoles() {\n    return this.http.get('https://localhost:5121/api/Admin').pipe(catchError(error => {\n      // Manejar el error aquí\n      console.error('Error en la solicitud:', error);\n      // Puedes lanzar el error nuevamente para que el observador lo maneje\n      return throwError(error);\n    }));\n  }\n}\nUserServices.ɵfac = function UserServices_Factory(t) {\n  return new (t || UserServices)(i0.ɵɵinject(i1.HttpClient));\n};\nUserServices.ɵprov = /*@__PURE__*/i0.ɵɵdefineInjectable({\n  token: UserServices,\n  factory: UserServices.ɵfac,\n  providedIn: 'root'\n});","map":{"version":3,"mappings":"AAGA,SAASA,UAAU,EAAEC,UAAU,QAAQ,MAAM;;;AAM7C,OAAM,MAAOC,YAAY;EAErBC,YAAoBC,IAAgB;IAAhB,SAAI,GAAJA,IAAI;EAAgB;EAExCC,QAAQ;IACJ,OAAO,IAAI,CAACD,IAAI,CAACE,GAAG,CAAU,kCAAkC,CAAC,CAC5DC,IAAI,CACDP,UAAU,CAACQ,KAAK,IAAG;MACf;MACAC,OAAO,CAACD,KAAK,CAAC,wBAAwB,EAAEA,KAAK,CAAC;MAC9C;MACA,OAAOP,UAAU,CAACO,KAAK,CAAC;IAC5B,CAAC,CAAC,CACL;EACT;;AAdSN,YAAY;mBAAZA,YAAY;AAAA;AAAZA,YAAY;SAAZA,YAAY;EAAAQ,SAAZR,YAAY;EAAAS,YAFT;AAAM","names":["catchError","throwError","UserServices","constructor","http","getRoles","get","pipe","error","console","factory","providedIn"],"sourceRoot":"","sources":["D:\\proyecto web c shrap\\Angular.net\\UserLogin\\login\\ClientApp\\src\\app\\services\\userServices.ts"],"sourcesContent":["import { Injectable } from '@angular/core';\r\nimport { HttpClient } from '@angular/common/http';\r\nimport { Roles } from '../Models/Roles.models';\r\nimport { catchError, throwError } from 'rxjs';\r\nimport { error } from 'console';\r\n\r\n@Injectable({\r\n    providedIn: 'root'\r\n})\r\nexport class UserServices {\r\n\r\n    constructor(private http: HttpClient) { }\r\n\r\n    getRoles() {\r\n        return this.http.get<Roles[]>('https://localhost:5121/api/Admin')\r\n            .pipe(\r\n                catchError(error => {\r\n                    // Manejar el error aquí\r\n                    console.error('Error en la solicitud:', error);\r\n                    // Puedes lanzar el error nuevamente para que el observador lo maneje\r\n                    return throwError(error);\r\n                })\r\n            );\r\n    }\r\n\r\n}\r\n"]},"metadata":{},"sourceType":"module","externalDependencies":[]}